### ISSUE TITLE
Anagram exercise is difficult for beginners

### ISSUE THREAD

**Issue by Gamecock** [exercism/c#180]
Extracting from #8
```
I think Anagram should be moved later in the track. A lot of people having trouble with the malloc.
That's a good observation. Sounds like a wise move. 
```
Two ways to solve this:

1. move it later in the track.  I need a hint on where to look for this.
2. Don't return a structure, and add a field to the input structure to mark if the word is an Anagram.  I can propose something here if we want to go this route.

**Comment by wolf99** [comment]
My 2c: It should be moved later in the track. The complexity of the `malloc()` is a nice step up from a very basic usage, while still not being massively obtuse, such that it requires a small but significant step in learning how to properly think about and use `malloc()`. The rest of the exercise is just string manipulation (maybe the `qsort()` is new), which people should be pretty comfortable by that stage as there are a couple of other exercises that do that.

**Comment by Gamecock** [comment]
If you look at the [config.json](https://github.com/exercism/c/blob/master/config.json) Anagram should be coming a lot later, after acronym.  Not sure where to look to figure this out.

**Comment by ryanplusplus** [comment]
@Gamecock unfortunately `config.json` is in a bit of a weird state due to the work that's being done towards Exercism v2. The next iteration of Exercism is going to have a different model for progression where there are some core exercises that can unlock non-core exercises that are similar or take the concepts in the core exercise further. Exercism v1 (the current version of the site), ignores the `unlocked_by` field and uses only the ordering in `config.json`. We can fix this just by moving the Anagram blob further down in the file, though that change will be made moot when v2 launches.

If you're interested in previewing the new site, you can check it out at https://v2.exercism.io/.

**Comment by ryanplusplus** [comment]
Closed by #182
